### PR TITLE
iday - Support external secret operator

### PR DESCRIPTION
**Pull request by bobotimi**
Allows secrets to be transparently synced from external secret providers such as AWS Secrets Manager, Azure Key Vault, Hashicorp Vault etc see [full list of providers](https://external-secrets.io/latest/provider/aws-secrets-manager/) )to Kubernetes secrets using the [external-secret operator](https://external-secrets.io/latest/)

### CODE REVIEW

**Comment by jarcilac** [comment]
@bobotimi, I was wondering if we should also include the new secrets file. since it should be treated as the license file.

**Review by jarcilac — APPROVED** [review]

**Review by Devleooper — APPROVED** [review]

File: nxrm-ha/values.yaml [hunk 1]
```diff
@@ -217,6 +220,44 @@ service:  #Nexus Repo NodePort Service
...
+#          serviceAccountRef:
+#            name: nexus-repository-deployment-sa # use same service account name as specified in serviceAccount.name
+  secrets:
+    database:
+      refreshInterval: 1h
+      valueIsJson: false
+      providerSecretName: dbSecretName # The name of the AWS SecretsManager/Azure KeyVault etc secret
+      dbUserKey: username #  the name of the key in the secret which contains your database username.
+      dbPasswordKey: password # the name of the key in the secret which contains your database password.
+      dbHostKey: host # the name of the key in the secret which contains your database host.
```
Comment (lisadurant) [inline nxrm-ha/values.yaml:251]:
```suggestion
      providerSecretName: dbSecretName # The name of the AWS SecretsManager/Azure KeyVault/etc. secret
      dbUserKey: username #  The name of the key in the secret that contains your database username
      dbPasswordKey: password # The name of the key in the secret that contains your database password
      dbHostKey: host # The name of the key in the secret that contains your database host
```

File: nxrm-ha/values.yaml [hunk 2]
```diff
@@ -217,6 +220,44 @@ service:  #Nexus Repo NodePort Service
...
+    admin:
+      refreshInterval: 1h
+      valueIsJson: false
+      providerSecretName: adminSecretName # The name of the AWS SecretsManager/Azure KeyVault etc secret
+      adminPasswordKey: "nexusAdminPassword" # the name of the key in the secret which contains your nexus repository admin password
```
Comment (lisadurant) [inline nxrm-ha/values.yaml:256]:
```suggestion
      providerSecretName: adminSecretName # The name of the AWS SecretsManager/Azure KeyVault/etc. secret
      adminPasswordKey: "nexusAdminPassword" # The name of the key in the secret that contains your nexus repository admin password
```

File: nxrm-ha/values.yaml [hunk 3]
```diff
@@ -217,6 +220,44 @@ service:  #Nexus Repo NodePort Service
...
+    license:
+      providerSecretName: nexus-repo-license.lic # The name of the AWS SecretsManager/Azure KeyVault etc secret which contains your Nexus Repository license
```
Comment (lisadurant) [inline nxrm-ha/values.yaml:258]:
```suggestion
      providerSecretName: nexus-repo-license.lic # The name of the AWS SecretsManager/Azure KeyVault/etc. secret that contains your Nexus Repository license
```

File: nxrm-ha/values.yaml [hunk 4]
```diff
@@ -217,6 +220,44 @@ service:  #Nexus Repo NodePort Service
...
+    name: nexus-secret-store
+    spec:
+      provider:
+#        aws:
+#          service: SecretsManager
+#          region: us-east-1
+#          auth:
+#            jwt:
+#              serviceAccountRef:
+#                name: nexus-repository-deployment-sa # use same service account name as specified in serviceAccount.name
```
Comment (lisadurant) [inline nxrm-ha/values.yaml:235]:
```suggestion
#                name: nexus-repository-deployment-sa # Use the same service account name as specified in serviceAccount.name
```

File: nxrm-ha/values.yaml [hunk 5]
```diff
@@ -217,6 +220,44 @@ service:  #Nexus Repo NodePort Service
...
+# Example for Azure
+#    spec:
+#      provider:
+#        azurekv:
+#          authType: WorkloadIdentity
+#          vaultUrl: "https://xx-xxxx-xx.vault.azure.net"
+#          serviceAccountRef:
+#            name: nexus-repository-deployment-sa # use same service account name as specified in serviceAccount.name
```
Comment (lisadurant) [inline nxrm-ha/values.yaml:243]:
```suggestion
#            name: nexus-repository-deployment-sa # Use the same service account name as specified in serviceAccount.name
```

**Review by lisadurant — CHANGES_REQUESTED** [review]

**Review by lisadurant — APPROVED** [review]

**Pull request merged**